src/transport/raw/TCP.cpp

```diff
@@ -616,36 +620,17 @@ CHIP_ERROR TCPBase::TCPConnect(const PeerAddress & address, Transport::AppTCPCon
 
 void TCPBase::TCPDisconnect(const PeerAddress & address)
 {
-    CHIP_ERROR err = CHIP_NO_ERROR;
     // Closes an existing connection
     for (size_t i = 0; i < mActiveConnectionsSize; i++)
     {
         if (mActiveConnections[i].IsConnected())
         {
-            Inet::IPAddress ipAddress;
-            uint16_t port;
-            Inet::InterfaceId interfaceId;
+            const Inet::IPAddress & ipAddress = mActiveConnections[i].mPeerAddr.GetIPAddress();
+            uint16_t port                     = mActiveConnections[i].mPeerAddr.GetPort();
 
-            err = mActiveConnections[i].mEndPoint->GetPeerInfo(&ipAddress, &port);
-            if (err != CHIP_NO_ERROR)
-            {
-                ChipLogError(Inet, "TCPDisconnect: GetPeerInfo error: %" CHIP_ERROR_FORMAT, err.Format());
-                return;
-            }
-
-            err = mActiveConnections[i].mEndPoint->GetInterfaceId(&interfaceId);
-            if (err != CHIP_NO_ERROR)
-            {
-                ChipLogError(Inet, "TCPDisconnect: GetInterfaceId error: %" CHIP_ERROR_FORMAT, err.Format());
-                return;
-            }
             // if (address == PeerAddress::TCP(ipAddress, port, interfaceId))
```

Comment:
This comment does not make sense: it does not match the code.

What we are actually doing here is ignoring the address type (TCP vs UDP) and ignoring the interface ID, right?  As in, this should be documenting why we are not in fact using `PeerAddress::operator==`.

Reply:
Needed to remove this. It was a lingering codeline from before. 
Yes, we are ignoring the InterfaceID because the IPAddress and Port are the minimal sufficient set of parameters for the connection. Furthermore, the interface information may not be there because it was not supplied during connection establishment, and so might pose an issue during comparison. Can add more comments on this to clarify.